### DISCUSSION TITLE
Yarn install fails with Next.js + Yarn workspace even with the official example

### DISCUSSION THREAD

**Discussion by arwtyxouymz** [vercel/community#1552] [Help] [answered]
# Summary

I'm trying to deploy my next.js app to vercel and I'm using yarn workspace (with turbo).
But during `yarn install` in the vercel build step, it fails with the error below.
```
Couldn't find any versions for "<package-name>" that matches "workspace:*"
```

I thought I configured it wrongly, but I found it fails with the official example!
Is it the bug or am I missing anything?

### Reproduce Step
1. Deploy this source to vercel
https://github.com/vercel/examples/tree/main/solutions/monorepo
2. It uses `pnpm` by default, so plz change to use `yarn` from the config
3. Redeploy it
4. You will see the error!
<img width="912" alt="スクリーンショット 2023-02-16 16 37 52" src="https://user-images.githubusercontent.com/16755660/219298791-40003f43-ffb8-40f4-aec0-23e80937f386.png">

```bash
error Couldn't find package "@acme/ui@workspace:*" required by "app@0.0.0" on the "npm" registry.
```

I guess I'm missing something, but I don't know what I'm missing.
Could you help me, plz?

**Comment by amyegan** [comment, marked as answer]
That example repo uses using Turborepo and [pnpm workspaces](https://pnpm.io/workspaces), so it won't be as simple as swapping out `yarn` for `pnpm` CLI. 

There is a [Turborepo with Yarn](https://github.com/vercel/turbo/tree/main/examples/with-yarn) example that I think will be better starting point for you. We also have a guide on [Deploying a Monorepo Using Yarn Workspaces to Vercel](https://vercel.com/guides/deploying-yarn-monorepos-to-vercel) that I find helpful 😃

**Reply by arwtyxouymz** [reply]
@amyegan 
Thank you for the reply!
I read that article but I couldn't resolve the issue...

Also, I tried `Turborepo with Yarn` example and I found `yarn install` fails if I use `workspace:*` range. 
This is the example of I changed turbo repo example from `*` to `workspace:*`.
https://github.com/arwtyxouymz/with-yarn-workspace-sample/commit/63ece0166e8373919077bd942dec5083c2451e12#diff-14b60f636e1a2b0061da57aaf231cb1ed15a5dc0c592425ed82e58fec95d42d8

When I use `*` for workspace dependencies, it works, but it fails when `workspace:*`.
Do you have any idea? I want to use `workspace:*` to promise to resolve workspace packages.

**Reply by frencojobs** [reply]
`workspace:version` convention is a feature of PNPM and Yarn v2 that doesn't exist in Yarn v1. For Yarn v1, you have to use `*` and you'll get mostly the same behavior.

**Reply by arwtyxouymz** [reply]
@frencojobs Ah that was the reason. I was using Yarn v3 in my project and I didn't know that [vercel doesn't support yarn v2 & v3](https://vercel.com/guides/does-vercel-support-yarn-2).

After I downgraded yarn v3 -> v1, it worked! Thank you!
